packages/eslint-plugin-template/src/rules/headings-consecutive-order.ts

```diff
@@ -0,0 +1,51 @@
+import type { TmplAstElement } from '@angular-eslint/bundled-angular-compiler';
+import { createESLintRule } from '../utils/create-eslint-rule';
+import { getTemplateParserServices } from '@angular-eslint/utils';
+
+export const RULE_NAME = 'headings-consecutive-order';
+export type MessageId = 'headingsOrder';
+
+export default createESLintRule<[], MessageId>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      // link: https://www.w3.org/WAI/tutorials/page-structure/headings/
+      description:
+        'Ensures that headings have consecutive order from largest to lowest h1 -> h6',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      headingsOrder: 'Headings number sequence must not be skipped',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+    let headingLastIndex: number;
+
+    return {
+      [`Element$1[name=/^h\\d/]`](element: TmplAstElement) {
```

Comment:
Might be good to filter this to only known heading elements similar to the `elements-content` rule:
```suggestion
      [`Element$1[name=/^(h1|h2|h3|h4|h5|h6)$/`](element: TmplAstElement) {
```